plasmapy_nei/eigen/eigenclass.py

```diff
@@ -0,0 +1,354 @@
+    ----------
+    element : particle-like
+        Representation of the element to access data for.
+
+    Examples
+    --------
+    >>> eigendata = EigenData("He")
+    >>> eigendata.nstates
+    3
+    """
+
+    def _validate_element(self, element):
+
+        # The following might not be needed if the check is in @particle_input
+        if not element.is_category(require="element", exclude=["ion", "isotope"]):
+            raise ValueError(f"{element} is not an element")
+
+        if element.atomic_number > max_atomic_number:
+            raise ValueError("Need an element")
+
+        self._element = element
+
+    def _load_data(self):
+        """
+        Retrieve data from the HDF5 file containing ionization and
+        recombination rates.
+        """
+        filename = pkg_resources.resource_filename(
+            "plasmapy_nei",
+            "data/ionrecomb_rate.h5",  # from Chianti database, version 8.07
+        )
+
+        # TODO: enable different HDF5 files
+
+        with h5py.File(filename, "r") as file:
+            self._temperature_grid = file["te_gird"][:]  # TODO: fix typo in HDF5 file
+            self._ntemp = len(self.temperature_grid)
+            c_ori = file["ioniz_rate"][:]
+            r_ori = file["recomb_rate"][:]
+
+        c_rate = np.zeros((self.ntemp, self.nstates))
+        r_rate = np.zeros((self.ntemp, self.nstates))
+        for temperature_index in range(self.ntemp):
+            for i in range(self.nstates - 1):
+                c_rate[temperature_index, i] = c_ori[
+                    i, self.element.atomic_number - 1, temperature_index
+                ]
+            for i in range(1, self.nstates):
+                r_rate[temperature_index, i] = r_ori[
+                    i - 1, self.element.atomic_number - 1, temperature_index
+                ]
+
+        self._ionization_rate = np.ndarray(
+            shape=(self.ntemp, self.nstates), dtype=np.float64
+        )
+        self._recombination_rate = np.ndarray(
+            shape=(self.ntemp, self.nstates), dtype=np.float64
+        )
+        self._equilibrium_states = np.ndarray(
+            shape=(self.ntemp, self.nstates), dtype=np.float64
+        )
+        self._eigenvalues = np.ndarray(
+            shape=(self.ntemp, self.nstates), dtype=np.float64
+        )
+
+        self._eigenvectors = np.ndarray(
+            shape=(self.ntemp, self.nstates, self.nstates), dtype=np.float64,
+        )
+
+        self._eigenvector_inverses = np.ndarray(
+            shape=(self.ntemp, self.nstates, self.nstates), dtype=np.float64,
+        )
+
+        self._ionization_rate[:, :] = c_rate[:, :]
+        self._recombination_rate[:, :] = r_rate[:, :]
+
+        # Define the coefficients matrix A. The first dimension is
+        # for elements, and the second number of equations.
+
+        nequations = self.nstates
+        coefficients_matrix = np.zeros(
+            shape=(self.nstates, nequations), dtype=np.float64
+        )
```

Comment:
I think storing this data in `xarray.Dataset` is a very good option.  However, I'd leave it to a separate PR, because I'd put all the object data into the xarray which would require restructuring of the class methods and properties.